### DISCUSSION TITLE
Network extraction from porespy leading to throats larger than adjacent pores

### DISCUSSION THREAD

**Discussion by Gal-Tzvik** [PMEAL/OpenPNM#2910] [Q&A]
hello, 
considering the snow2 algorithm and the network extraction, i am getting a network with throats larger than adjacent pores. therefore some geometric modules cant function. i tried looking at how the snow2 algorithm defines the throats and their sizes in the code but i cant fully understand. are there any references i can look at?

thank you,
Gal

**Comment by Gal-Tzvik** [comment]
Additionally, is it possible that using the snow2 algorithm with high accuracy, the diameter is calculated based on surface area and not cross sectional area?

**Reply by jgostick** [reply]
If your pores are a bit 'flat', then the spheres will be small, but the throats connected to the wide side can be large.  The spheres and cones model is able to handle this.  Or you can artificially reduce throat sizes.

**Reply by Gal-Tzvik** [reply]
I assume you are referring to the cones and cylinder collection, Although i dont see any cones model in the independent geometrical models. My problem with using a geometrical collection model is that is automatically generates a pore seed model and therefore altering my actual pore values. I didnt find a way you delete this model or concel the dependancy of other models on the pore.seed model.

**Reply by jgostick** [reply]
You can delete the models you don't want, like `del pn.models['pore.seed@all']`

**Reply by Gal-Tzvik** [reply]
Awesome! the Cones and cylinders seems to have done the job. If possible I still have many questions regarding the geometrical property extraction algorithms. I will organize my exact questions and hopefully you will have the time. 
Thanks!

**Comment by likeduck** [comment]
This is likely due to the input image being too complex.
You can try to modify 'r_max' and 'sigma'.
`regions = ps.filters.snow_partitioning(im=im, r_max=r_max, sigma=sigma)`

**Reply by Gal-Tzvik** [reply]
I will evaluate this..thank you
